src/redux/actions/experiments/createExperiment.js

```diff
@@ -1,47 +1,99 @@
 /* eslint-disable no-param-reassign */
+import fetchAPI from 'utils/fetchAPI';
 import moment from 'moment';
 import hash from 'object-hash';
 
-import saveExperiment from './saveExperiment';
+import { api } from 'utils/constants';
+
 import {
   EXPERIMENTS_CREATED,
   EXPERIMENTS_ERROR,
-} from '../../actionTypes/experiments';
-import { experimentTemplate } from '../../reducers/experiments/initialState';
-import endUserMessages from '../../../utils/endUserMessages';
-import pushNotificationMessage from '../../../utils/pushNotificationMessage';
+  EXPERIMENTS_SAVING,
+  EXPERIMENTS_SAVED,
+} from 'redux/actionTypes/experiments';
+import { experimentTemplate } from 'redux/reducers/experiments/initialState';
+
+import endUserMessages from 'utils/endUserMessages';
+import pushNotificationMessage from 'utils/pushNotificationMessage';
+import { isServerError, throwIfRequestFailed } from 'utils/fetchErrors';
+import experimentConvertedToApiModel from 'utils/convertExperimentToApiModel';
 
 const createExperiment = (
   projectUuid, newExperimentName,
-) => async (dispatch) => {
+) => async (dispatch, getState) => {
   const createdDate = moment().toISOString();
 
+  const experimentId = hash.MD5(createdDate);
+
   const newExperiment = {
     ...experimentTemplate,
-    id: hash.MD5(createdDate),
+    id: experimentId,
     name: newExperimentName,
     projectUuid,
     createdDate,
   };
 
+  let url;
+  let experimentToSend;
+
   try {
```

Comment:
Why do we need the try block to begin here? Are we expecting errors during the conversion of the models? Otherwise I think the try block can begin much later in the code, just after dispatching experiments saving.